app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt

```diff
@@ -0,0 +1,153 @@
+import com.mapbox.maps.plugin.gestures.gestures
+import com.mapbox.maps.testapp.BaseMapTest
+import com.mapbox.maps.testapp.test.R
+import com.mapbox.maps.viewannotation.OnViewAnnotationUpdatedListener
+import com.mapbox.maps.viewannotation.viewAnnotationOptions
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+/**
+ * Tests gestures intercepted on view annotations.
+ */
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class GestureInterceptedOnViewAnnotationTest : BaseMapTest() {
+
+  private fun withLatch(
+    count: Int = 1,
+    timeout: Long = 3000,
+    lambda: (CountDownLatch) -> Unit
+  ) {
+    val latch = CountDownLatch(count)
+    lambda(latch)
+    if (!latch.await(timeout, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  @Test
+  fun clickableViewAnnotationDoesNotTriggerLongTouchAfterwards() {
+    val cameraCenter = Point.fromLngLat(-122.44121, 37.76132)
+
+    var longClicks = 0
+
+    withLatch { latch ->
+      rule.scenario.onActivity {
+        it.runOnUiThread {
+          mapView.gestures.addOnMapLongClickListener {
+            longClicks++
+            return@addOnMapLongClickListener true
+          }
+
+          mapboxMap.setCamera(
+            cameraOptions {
+              center(cameraCenter)
+            }
+          )
+
+          mapView.viewAnnotationManager.apply {
+
+            // xml layout resource is only available within instrumentationContext,
+            // while mapView uses targetContext hence fails to inflate if used via resId
+            val viewAnnotation = LayoutInflater.from(InstrumentationRegistry.getInstrumentation().context).inflate(R.layout.view_annotation, mapView, false)
+
+            addViewAnnotation(
+              view = viewAnnotation,
+              options = viewAnnotationOptions {
+                geometry(cameraCenter)
+              }
+            )
+
+            viewAnnotation.setOnClickListener { _ ->
+              removeViewAnnotation(viewAnnotation)
+            }
+
+            addOnViewAnnotationUpdatedListener(
+              object : OnViewAnnotationUpdatedListener {
+                override fun onViewAnnotationPositionUpdated(
+                  view: View,
+                  leftTopCoordinate: ScreenCoordinate,
+                  width: Int,
+                  height: Int
+                ) {}
+
+                override fun onViewAnnotationVisibilityUpdated(view: View, visible: Boolean) {
+                  if (view == viewAnnotation && visible) {
+                    latch.countDown()
+                  }
+                }
+              }
+            )
+          }
+        }
+      }
+    }
+
+    // click on view annotation to close it
+    onView(withId(R.id.view_annotation)).perform(
+      GesturesUiTestUtils.move(
```

Comment:
just a question
click made with move action?

Reply:
Yup, move delta is small so it's effectively a click :)

Reply:
also you can add some extension: 

```
R.id.view_annotation.click()
```

```
com.mapbox.maps.R.id.mapView.move(deltaX, deltaY)
```
and hide `onVIew(withId(...)).perform(...`

Reply:
Added `click()` shorthand for move with arguments, however I wouldn't change the existing calls to extensions of Int, could be confusing and increases number of change if applied accross the codebase